src/Chainweb/Miner/Coordinator.hs

```diff
@@ -167,6 +164,54 @@ filterAdjustments newBh as = case window $ _blockChainwebVersion newBh of
             limit = bool (_blockHeight newBh - wh) 0 (_blockHeight newBh < wh)
         in HM.filter (\(T2 h _) -> h > limit) as
 
+-- | THREAD: Accepts a "solved" `BlockHeader` from some external source (likely
+-- a remote mining client), reassociates it with the `Cut` from which it
+-- originated, and publishes it to the `Cut` network.
+--
+publishing :: LogFunction -> TVar (Maybe MiningState) -> CutDb cas -> BlockHeader -> IO ()
+publishing lf tp cdb bh = do
+    -- Reassociate the new `BlockHeader` with the current `Cut`, if possible.
+    -- Otherwise, return silently.
+    --
+    c <- _cut cdb
+    readTVarIO tp >>= \case
+        Nothing -> pure ()  -- TODO Throw error?
+        Just (MiningState pl p) -> when (compatibleCut c && samePayload pl) $ do
```

Comment:
I assume that `samePayload` fails rarely enough, so that it doesn't pay of to cache the payload and cache in the rewards, even if the payload is not the most recent?